pdns/lua-record.cc

```diff
@@ -508,7 +517,7 @@ static vector<pair<int, ComboAddress> > convWIplist(std::unordered_map<int, wipl
   vector<pair<int,ComboAddress> > ret;
 
   for(const auto& i : src) {
-    ret.emplace_back(atoi(i.second.at(1).c_str()), ComboAddress(i.second.at(2)));
+    ret.emplace_back(pdns_stou_nonzero(i.second.at(1).c_str()), ComboAddress(i.second.at(2)));
```

Comment:
shouldn't the vector pair be unsigned int too?

Reply:
I ignored that at first because people shouldn't be using weights that care about that, but it would be safer, I agree. Will fix.